### PR TITLE
Create virtual network template

### PR DESCRIPTION
**Pull request by weeyin83**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by bmoore-msft** [comment]
Many of the samples in the repo deploy a vnet, is there something different we want this sample to do?

**Comment by weeyin83** [comment]
Hi @bmoore-msft , I created this template to allow users to start from the ground up in setting up their Azure infrastructure.  Alot of customer sites I go into we start off simple by setting up the network side of things within Azure then deploy VMs, etc on top of that.  The current templates are great but they bundle alot of elements together all at once, this template I have come up with allows users to start small and get a handle on designing and deploying to Azure.  Maybe an update to the readme file would help explain this concept better?

File: 101-virtual-network/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Create a Virtual Network
```
Comment (bmoore-msft) [inline 101-virtual-network/README.md]: update the readme with a little more detail about why this sample is useful

File: 101-virtual-network/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "type": "string",
+        "metadata": {
+            "description": "Virtual Network Name"
+        }
+        },
+        "addressPrefix (i.e. 10.0.0.0/24)": {
```
Comment (bmoore-msft) [inline 101-virtual-network/azuredeploy.json]: remove the (ie...) from the param name - use description for explanations

File: 101-virtual-network/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
```
Comment (bmoore-msft) [inline 101-virtual-network/azuredeploy.json]: use something more descriptive than "name" - e.g. virtualNetworkName

File: 101-virtual-network/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        },
+        "subnetName": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Subnet Name"
+            }
+        },
+      "subnetAddressPrefix (i.e. 10.0.0.0/24)": {
```
Comment (bmoore-msft) [inline 101-virtual-network/azuredeploy.json]: same as above

File: 101-virtual-network/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        },
+        "addressPrefix (i.e. 10.0.0.0/24)": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description":"Address Space"
+            }
+
+        },
+        "subnetName": {
```
Comment (bmoore-msft) [inline 101-virtual-network/azuredeploy.json]: move to variable

File: 101-virtual-network/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    "parameters": {
+        "name": {
+            "type": "string",
+        "metadata": {
+            "description": "Virtual Network Name"
+        }
+        },
+        "addressPrefix (i.e. 10.0.0.0/24)": {
+            "type": "string",
+            "defaultValue": "",
```
Comment (bmoore-msft) [inline 101-virtual-network/azuredeploy.json]: use valid defaultValues

File: 101-virtual-network/azuredeploy.parameters.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "name": {
+      "value": "vnet"
+    },
+    "addressPrefix (i.e. 10.0.0.0/24)": {
```
Comment (bmoore-msft) [inline 101-virtual-network/azuredeploy.parameters.json]: use defaultValues where possible

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**